drivers/gpu/drm/vc4/vc4_firmware_kms.c

```diff
@@ -958,6 +953,16 @@ static void vc4_crtc_mode_set_nofb(struct drm_crtc *crtc)
 			if (conn_state->broadcast_rgb ==
 						VC4_BROADCAST_RGB_LIMITED)
 				mb.timings.flags |= TIMINGS_FLAGS_RGB_LIMITED;
+
+			/* If not using the default range, then do not provide
+			 * a VIC as the HDMI spec requires that we do not
+			 * signal the opposite of the defined range in the AVI
+			 * infoframe.
+			 */
+			if (!!(mb.timings.flags & TIMINGS_FLAGS_RGB_LIMITED) ^
+			    (drm_default_rgb_quant_range(mode) ==
+					HDMI_QUANTIZATION_RANGE_LIMITED))
```

Comment:
This logic is a bit unclear.

Reply:
If the `^` where a `!=` would that be clearer?

Reply:
Is that a request for a change, or an explanation for JamesH65?

Reply:
To change, or not to change? That is the question.

Reply:
`^` has no benefit over `!=` here, so change it.

Reply:
Done.